src/fsharp/FSharp.Core/local.fs

```diff
@@ -72,6 +72,19 @@ module internal List =
             let cons = freshConsNoTail x
             distinctByToFreshConsTail cons hashSet keyf rest
             cons
+    
+    let countBy (dict:Dictionary<_, int>) (keyf:'T -> 'Key) = 
+        use mutable ie = dict.GetEnumerator()
```

Comment:
Alright I changed it to a let. Is there a reason we don't need to dispose the dictionary enumerator, but that we need to dispose general enumerators?